rosidl_generator_type_description/rosidl_generator_type_description/__init__.py

```diff
@@ -384,7 +394,7 @@ def field_type_type_name(ftype: definition.AbstractType) -> str:
 
     if isinstance(value_type, definition.BasicType):
         value_type_name = FIELD_VALUE_TYPE_NAMES[value_type.typename]
-    elif isinstance(value_type, definition.AbstractGenericString):
+    elif isinstance(value_type, FIELD_VALUE_STRING_TYPES):
```

Comment:
I will point out that this is not *quite* the same thing as what it is replacing.  In particular, while `FIELD_VALUE_STRING_TYPES` covers all of the currently defined `AbstractGenericString` types, it does not account for any types that might be derived in the future, or in user code.  While I think those are unlikely, they are not impossible.

Can you explain a bit more why you did this change?

Reply:
Since this value is used to index into `FIELD_VALUE_TYPE_NAMES`. It has the more specific sub classes. If you wanted to add future string types you would need to update `FIELD_VALUE_STRING_TYPES` as well as `FIELD_VALUE_TYPE_NAMES`.

```python
FIELD_VALUE_TYPE_NAMES: Final = {
    None: 'FIELD_TYPE_NOT_SET',
    'nested_type': 'FIELD_TYPE_NESTED_TYPE',
    'int8': 'FIELD_TYPE_INT8',
    'uint8': 'FIELD_TYPE_UINT8',
    'int16': 'FIELD_TYPE_INT16',
    'uint16': 'FIELD_TYPE_UINT16',
    'int32': 'FIELD_TYPE_INT32',
    'uint32': 'FIELD_TYPE_UINT32',
    'int64': 'FIELD_TYPE_INT64',
    'uint64': 'FIELD_TYPE_UINT64',
    'float': 'FIELD_TYPE_FLOAT',
    'double': 'FIELD_TYPE_DOUBLE',
    'long': 'LONG_DOUBLE',
    'char': 'FIELD_TYPE_CHAR',
    'wchar': 'FIELD_TYPE_WCHAR',
    'boolean': 'FIELD_TYPE_BOOLEAN',
    'octet': 'FIELD_TYPE_BYTE',
    definition.UnboundedString: 'FIELD_TYPE_STRING',
    definition.UnboundedWString: 'FIELD_TYPE_WSTRING',
    # NOTE: rosidl_parser does not define fixed string types
    definition.BoundedString: 'FIELD_TYPE_BOUNDED_STRING',
    definition.BoundedWString: 'FIELD_TYPE_BOUNDED_WSTRING',
}
```